zipline/finance/controls.py

```diff
@@ -269,6 +269,29 @@ def validate(self,
             self.fail(sid, amount)
 
 
+class ExpiredSID(TradingControl):
+    """
+    TradingControl representing a prohibition against ordering a SID that has
+    passed its end_date.
+    """
+
+    def validate(self,
+                 sid,
+                 amount,
+                 portfolio,
+                 algo_datetime,
+                 algo_current_data):
+        """
+        Fail if the algo has passed this Asset's end_date.
+        """
+        # No failure if the Asset has no end_date
+        if not sid.end_date:
```

Comment:
I refactored to your example. The expected empty value is a None - should I change it to a "if sid.end_date is None"?